other_langs/tests_cpp/hello_s3.cpp

```diff
@@ -0,0 +1,142 @@
+
+    auto outcome = client.ListBuckets();
+
+    bool result = true;
+    if (!outcome.IsSuccess()) {
+        std::cerr << "Failed with error: " << outcome.GetError() << std::endl;
+        result = false;
+    } else {
+        std::cout << "Found " << outcome.GetResult().GetBuckets().size() << " buckets\n";
+        for (auto &&b: outcome.GetResult().GetBuckets()) {
+            std::cout << b.GetName() << std::endl;
+        }
+    }
+
+    return result;
+}
+
+bool copyObject(const Aws::String &objectKey, const Aws::String &fromBucket, const Aws::String &toBucket,
+                            const Aws::S3::S3ClientConfiguration &clientConfig) {
+    Aws::S3::S3Client client(clientConfig);
+    Aws::S3::Model::CopyObjectRequest request;
+
+    request.WithCopySource(fromBucket + "/" + objectKey)
+            .WithKey(objectKey)
+            .WithBucket(toBucket);
+
+    Aws::S3::Model::CopyObjectOutcome outcome = client.CopyObject(request);
+    if (!outcome.IsSuccess()) {
+        const Aws::S3::S3Error &err = outcome.GetError();
+        std::cerr << "Error: copyObject: " <<
+                  err.GetExceptionName() << ": " << err.GetMessage() << std::endl;
+
+    } else {
+        std::cout << "Successfully copied " << objectKey << " from " << fromBucket <<
+                  " to " << toBucket << "." << std::endl;
+    }
+
+    return outcome.IsSuccess();
+}
+
+
+int main(int argc, char **argv) {
+    Aws::SDKOptions options;
+    // Optionally change the log level for debugging.
+//   options.loggingOptions.logLevel = Utils::Logging::LogLevel::Debug;
+    Aws::InitAPI(options); // Should only be called once.
+    int result = 0;
+    {
+        Aws::Client::ClientConfiguration clientConfig;
+        // Optional: Set to the AWS Region (overrides config file).
+        // clientConfig.region = "us-east-1";
+        clientConfig.endpointOverride = "http://localhost:5000";
+
+        // You don't normally have to test that you are authenticated. But the S3 service permits anonymous requests, thus the s3Client will return "success" and 0 buckets even if you are unauthenticated, which can be confusing to a new user.
+        auto provider = Aws::MakeShared<DefaultAWSCredentialsProviderChain>("alloc-tag");
+        auto creds = provider->GetAWSCredentials();
+
+        Aws::S3::S3Client s3Client(clientConfig);
+        listBuckets(clientConfig);
+
+        Aws::String uuid1 = Aws::Utils::UUID::RandomUUID();
+        Aws::String uuid2 = Aws::Utils::UUID::RandomUUID();
+        Aws::String bucket1 = Aws::Utils::StringUtils::ToLower(uuid1.c_str());
+        Aws::String bucket2 = Aws::Utils::StringUtils::ToLower(uuid2.c_str());
+
+        createBucket(bucket1, clientConfig);
+        createBucket(bucket2, clientConfig);
+
+        putObject(bucket1, "test.txt", clientConfig);
+
+        copyObject("test.txt", bucket1, bucket2, clientConfig);
+    }
+
+    Aws::ShutdownAPI(options); // Should only be called once.
+    return result;
```

Comment:
This returns 0 (success) even if errors occur.

Reply:
Great point - I validated manually that the test passed, but that's obviously not enough.

Appreciate the review!